src/tree.rs

```diff
@@ -217,41 +520,267 @@ impl PartialEq for Tree {
 /// would take one hash map lookup *per child*.
 #[derive(Debug)]
 struct Entry {
-    parent_index: Option<usize>,
     item: Item,
-    level: i64,
-    is_syncable: bool,
-    child_indices: Vec<usize>,
+    divergence: Divergence,
+    parents: EntryParents,
+    child_indices: Vec<Index>,
+}
+
+impl Entry {
+    fn root(root: Item) -> Entry {
+        Entry {
+            item: root,
+            divergence: Divergence::Ok,
+            parents: EntryParents::Root,
+            child_indices: Vec::new(),
+        }
+    }
+
+    fn is(&self, other: &Entry) -> bool {
+        self as *const _ == other as *const _
```

Comment:
It's not clear to me why this is a pointer comparison vs the fields deriving PartialEq (can you have dupe guids but don't want them to match?), can you clarify w/ a comment?